scrapy_zyte_smartproxy/middleware.py

```diff
@@ -265,9 +265,6 @@ def process_response(self, request, response, spider):
         if not self._is_enabled_for_request(request):
             return self._handle_not_enabled_response(request, response)
 
-        if not self._is_zyte_smartproxy_response(response):
-            return response
```

Comment:
Since `self._is_enabled_for_request(request)` is enough for `process_request`, I don’t see why we would need an additional check on the response.

I was originally going to check `request.meta["proxy"]`, but then I realized the plugin is also meant to work if you use multiple SPM/Zyte API proxy endpoints by setting them on that meta key manually.

Reply:
Thanks @Gallaecio ! I once had the similar doubt when creating this PR. However my motivation here was to limit the scope of this PR and quickly deliver a minor fix to the response check, instead of dropping the response check at all, which might better belong to a separate task that require further reviews and further discussions.

Now if I'm asked to give my two cents: I'd assume it's better that we keep the response check. At least it won't hurt (well, when this fix is applied).

Given my understanding of Scrapy's model, it may not be considered appropriate if we assume a response here comes from the SPM/ZAPI server (or comes from the network at all) even if the `_is_enabled_for_request` check passes.

One example is the use of HTTP cache. This SPM middleware is recommended to be installed at depth 610 while the builtin `HttpCacheMiddleware` at 900. Although the builtin implement does take response headers into consideration (so that the existing `_is_zyte_smartproxy_response` check fails as well), we cannot yet rule out the possibility that the cached data was originally acquired from a different source (thus it comes without the response headers). In this case we might want this additional response check to have its impact.